pipelined/bevy_pbr2/src/wireframe.rs

```diff
@@ -0,0 +1,144 @@
+    view::{ExtractedView, Msaa},
+    RenderApp, RenderStage,
+};
+use wgpu::PolygonMode;
+
+pub const WIREFRAME_SHADER_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 192598014480025766);
+
+#[derive(Debug, Default)]
+pub struct WireframePlugin;
+
+impl Plugin for WireframePlugin {
+    fn build(&self, app: &mut bevy_app::App) {
+        let mut shaders = app.world.get_resource_mut::<Assets<Shader>>().unwrap();
+        shaders.set_untracked(
+            WIREFRAME_SHADER_HANDLE,
+            Shader::from_wgsl(include_str!("render/wireframe.wgsl")),
+        );
+
+        app.init_resource::<WireframeConfig>();
+
+        app.sub_app(RenderApp)
+            .add_render_command::<Opaque3d, DrawWireframes>()
+            .init_resource::<WireframePipeline>()
+            .init_resource::<SpecializedPipelines<WireframePipeline>>()
+            .add_system_to_stage(RenderStage::Extract, extract_wireframes)
+            .add_system_to_stage(RenderStage::Extract, extract_wireframe_config)
+            .add_system_to_stage(RenderStage::Queue, queue_wireframes);
+    }
+}
+
+fn extract_wireframe_config(mut commands: Commands, wireframe_config: Res<WireframeConfig>) {
+    if wireframe_config.is_added() || wireframe_config.is_changed() {
+        commands.insert_resource(wireframe_config.into_inner().clone());
+    }
+}
+
+fn extract_wireframes(mut commands: Commands, query: Query<Entity, With<Wireframe>>) {
+    for entity in query.iter() {
+        commands.get_or_spawn(entity).insert(Wireframe);
+    }
+}
+
+/// Controls whether an entity should rendered in wireframe-mode if the [WireframePlugin] is enabled
+#[derive(Component, Debug, Clone, Default, Reflect)]
+#[reflect(Component)]
+pub struct Wireframe;
+
+#[derive(Debug, Clone, Default)]
+pub struct WireframeConfig {
+    /// Whether to show wireframes for all meshes. If `false`, only meshes with a [Wireframe] component will be rendered.
+    pub global: bool,
+}
+
+pub struct WireframePipeline {
+    mesh_pipeline: MeshPipeline,
+    shader: Handle<Shader>,
+}
+impl FromWorld for WireframePipeline {
+    fn from_world(render_world: &mut World) -> Self {
+        WireframePipeline {
+            mesh_pipeline: render_world.get_resource::<MeshPipeline>().unwrap().clone(),
+            shader: WIREFRAME_SHADER_HANDLE.typed(),
+        }
+    }
+}
+
+impl SpecializedPipeline for WireframePipeline {
+    type Key = MeshPipelineKey;
+
+    fn specialize(
+        &self,
+        key: Self::Key,
+    ) -> bevy_render2::render_resource::RenderPipelineDescriptor {
+        let mut descriptor = self.mesh_pipeline.specialize(key);
+        descriptor.vertex.shader = self.shader.clone_weak();
+        descriptor.fragment.as_mut().unwrap().shader = self.shader.clone_weak();
+        descriptor.primitive.polygon_mode = PolygonMode::Line;
+        descriptor.depth_stencil.as_mut().unwrap().bias.slope_scale = 1.0;
```

Comment:
I have no idea if `1.0` is a good value here but it looks way better than without.

![grafik](https://user-images.githubusercontent.com/22177966/143569892-1b5eabb2-ec0d-43f1-ab50-572bae36bed2.png)
![grafik](https://user-images.githubusercontent.com/22177966/143569958-5426bbad-7a38-4d03-8cc7-5062c99ae996.png)